src/treeffuser/_base_tabular_diffusion.py

```diff
@@ -97,41 +99,137 @@ def get_new_score_model(self) -> ScoreModel:
         Return the score model.
         """
 
-    def _validate_data(
+    def _preprocess_and_validate_data(
         self,
-        X: Optional[ndarray] = None,
-        y: Optional[ndarray] = None,
+        X: Optional[Union[ndarray, pd.DataFrame]] = None,
+        y: Optional[Union[ndarray, pd.Series]] = None,
```

Comment:
I don't know if you did this intentionally but I appreciate the Union for older python support :))

Reply:
it was intentional indeed :) as suggested by @aagrande in the past.